consensus/src/pipeline/virtual_processor/processor.rs

```diff
@@ -137,10 +356,31 @@ impl VirtualStateProcessor {
     }
 
     pub fn process_genesis_if_needed(self: &Arc<Self>) {
-        if let Some(BlockStatus::StatusUTXOValid) = self.statuses_store.read().get(self.genesis_hash).unwrap_option() {
-            return;
+        let status = self.statuses_store.read().get(self.genesis_hash).unwrap();
+        match status {
+            StatusUTXOPendingVerification => {
+                // TODO: consider using a batch write
+                self.virtual_state_store
+                    .write()
+                    .set(VirtualState::from_genesis(
+                        self.genesis_hash,
+                        self.ghostdag_manager.ghostdag(&[self.genesis_hash]).as_ref().clone(),
+                    ))
+                    .unwrap();
+                self.commit_utxo_state(self.genesis_hash, UtxoDiff::default(), MuHash::new(), AcceptanceData {});
+                match self.past_pruning_points_store.insert(0, self.genesis_hash) {
```

Comment:
If you flip the order there's no need to check `KeyAlreadyExists` (you can just unwrap), and there's no need for a batch

Reply:
There might be a case where it was written even though status was not updated. So we can't just unwrap since it will panic on existing key